customer-managed/azure/README.md

```diff
@@ -0,0 +1,271 @@
+These resources should be created in advance by the customer and then provided to Redpanda during cluster creation.
+The code is provided in [Terraform](https://developer.hashicorp.com/terraform) and Azure CLI (TODO).
+
+> There may be resources in this repository that already exist within your environment (for example, the VNET) that you don't want to create. Variables are provided for this purpose.
+
+# Customer Managed Resources
+Customer Managed Resources can be broken down into the following groups. You can walk through the code to get the exact list of the resources required to create and deploy Redpanda cluster.
+- Resource Group Resources
+- User Assigned Identities
+- IAM Roles and Assignments
+- Network
+- Storage
+- Key Vaults
+
+# Deploy Customer Managed Resources With Terraform
+## Prerequisites
+
+1. Access to an Azure subscription where you want to create your cluster.
+2. Knowledge of your internal VNET and subnet configuration.
+3. Permission to create, modify, and delete the resources described by this Terraform.
+4. [Terraform](https://developer.hashicorp.com/terraform/install) version 1.8.5 or later.
+
+## Setup
+
+> You may want to configure [remote state](https://developer.hashicorp.com/terraform/language/state/remote) for this
+> Terraform. For simplicity, these instructions assume local state.
+
+### Configure The Variables
+
+The variable files contain the variables that allow you to modify to meet your specific needs.
+- `vars.azure.tf`:  It contains the variables related to Azure credentials.
+- `vars.condition.tf`: It contains the conditional variables on whether to e.g. create VNET.
+- `vars.customer_input.tf`: It contains the variables needed for creating a cluster, which are required when using the Cloud API to create one.
+- `vars.iam.tf`: It contains the variables related to IAM roles.
+- `vars.misc.tf`: It contains the variables, region, zones, tags and resource name prefix.
+
+You can get the description of each variable in [here](./terraform/README.md).
+
+Create a JSON file called `byovnet.auto.tfvars.json` inside the Terraform directory.
+
+```shell
+{
+  "azure_tenant_id": "",
+  "azure_subscription_id": "",
+  "region": "",
+  "resource_name_prefix": "",
+  "tags": {},
+  "redpanda_resource_group_name": "",
+  "redpanda_storage_resource_group_name": "",
+  "redpanda_network_resource_group_name": "",
+  "redpanda_iam_resource_group_name": "",
+  "redpanda_agent_identity_name": "",
+  "redpanda_external_dns_identity_name": "",
+  "redpanda_cluster_identity_name": "",
+  "aks_identity_name": "",
+  "redpanda_console_identity_name": "",
+  "kafka_connect_identity_name": "",
+  "redpanda_management_storage_account_name": "",
+  "redpanda_management_storage_container_name": "",
+  "redpanda_tiered_storage_account_name": "",
+  "redpanda_tiered_storage_container_name": "",
+  "redpanda_management_key_vault_name": "",
+  "redpanda_console_key_vault_name": "",
+  "vnet_name": "",
```

Comment:
nit: there is a dependency between `vnet_name` and `create_vnet`. If `create_vnet` is false, then `vnet_name` is required. It's nice to avoid these where possible.